src/rulesets/LinearAlgebra/structured.jl

```diff
@@ -98,27 +98,31 @@ end
 Adjoint_mat_pullback(ȳ::Tangent) = (NoTangent(), ȳ.parent)
 Adjoint_mat_pullback(ȳ::AbstractVecOrMat) = (NoTangent(), adjoint(ȳ))
 Adjoint_mat_pullback(ȳ::AbstractThunk) = return Adjoint_mat_pullback(unthunk(ȳ))
+Adjoint_mat_pullback(ȳ::AbstractZero) = (NoTangent(), ȳ)
```

Comment:
Just an FYI: we are considering handling this at the AD level, see: https://github.com/JuliaDiff/ChainRules.jl/issues/442#issuecomment-872098608

But that is currently blocked by [`ProjectTo` PR](https://github.com/JuliaDiff/ChainRulesCore.jl/pull/385) which I am working on at the moment.

Reply:
Should these lines be removed before this is merged? Or commented `# remove me once X`?

Reply:
Ideally removed, but it needs Diffractor to automatically propagate AbstractZeros, something like Zygote's https://github.com/FluxML/Zygote.jl/blob/5b803d12f2d235755ba6f23a9017fc29fefd9985/src/compiler/chainrules.jl#L143

If Diffractor does not do that, we should make an issue about it (and link it here)

Reply:
So is it fine to leave this in or should I just remove that part for now?

Reply:
Yeah let's make sure it all works for Diffractor and just link this: https://github.com/JuliaDiff/Diffractor.jl/issues/25